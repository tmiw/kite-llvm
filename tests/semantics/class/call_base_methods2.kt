class X [
    construct(name) [ "exec X"|print; this.n = name; ] 
];

class Y from X [
    construct(name) [ "exec Y"|print; base|__init__(name); ]
];

class Z from Y [
    construct(name) [ "exec Z"|print; base|__init__(name); ]
];

v = make Z("xxx");
v.n|print;
