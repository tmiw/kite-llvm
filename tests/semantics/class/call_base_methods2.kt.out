exec Z
exec Y
exec X
xxx
